src/util/error-handler.ts

```diff
@@ -151,3 +151,146 @@ export const reactionRolesErrorHandler: ErrorHandler = async (ctx: ButtonInterac
     
     // ----- !ERROR LOG -----
 };
+
+/**
+ * Actions related to the members logs category.
+ */
+export enum MemberLogsAction {
+    MemberJoin = "Member Join",
+    MemberLeave = "Member Leave",
+    MemberBan = "Member Ban",
+    MemberUnban = "Member Unban"
+}
+
+/**
+ * Actions related to the profiles logs category.
+ */
+export enum ProfileLogsAction {
+    MemberUpdate = "Member Update",
+    UserUpdate = "User Update"
+}
+
+/**
+ * Actions related to the messages logs category.
+ */
+export enum MessageLogsAction {
+    MessageEdit = "Message Edit",
+    MessageDelete = "Message Delete"
+}
+
+/**
+ * All logs categories.
+ */
+type LogsAction = MemberLogsAction |
+    ProfileLogsAction |
+    MessageLogsAction;
+
+/**
+ * The trigger of a logs error.
+ * A User if it's related to a user update, or a ban, or
+ * a message.
+ * otherwise a member.
+ */
+type Trigger<T extends LogsAction> = T extends ProfileLogsAction.UserUpdate |
+    MemberLogsAction.MemberBan |
+    MemberLogsAction.MemberUnban |
+    MessageLogsAction ? User : GuildMember;
+
+/**
+ * The payload of a logs error.
+ * The before and after states of a Message if it's a message edit action, or
+ * a single message if it's a message delete action, or
+ * the before and after states of a User if it's a user update action, or
+ * the before and after states of a Member if it's a member update action.
+ */
+type Payload<T extends LogsAction> = T extends MessageLogsAction.MessageEdit ? [before: Message, after: Message] :
+    T extends MessageLogsAction ? [message: Message] :
+    T extends ProfileLogsAction.UserUpdate ? [before: User, after: User] :
+    T extends ProfileLogsAction.MemberUpdate ? [before: GuildMember, after: GuildMember] :
+    [];
+
+/**
+ * Handles errors related to moderation logs.
+ * It logs the error along with any payload to the error logs channel.
+ * 
+ * @param action the logs action
+ * @param server the server the error originated from
+ * @param trigger the member or user who triggered the error
+ * @param err the error that occurred
+ * @param payload the error payload
+ */
+export const moderationLogsErrorHandler = async <T extends LogsAction>(
+    action: T,
+    server: Guild,
+    trigger: Trigger<T>,
+    err: Error,
+    ...payload: Payload<T>
+): Promise<void> => {
```

Comment:
This is the error handler for moderation logs errors. It uses conditional types for its parameters.